plugins/plugin_utils/unsafe.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2023, Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+
+from ansible.module_utils.six import binary_type, text_type
+from ansible.module_utils.common._collections_compat import Mapping, Set
+from ansible.module_utils.common.collections import is_sequence
+from ansible.utils.unsafe_proxy import (
+    AnsibleUnsafe,
+    wrap_var as _make_unsafe,
+)
+
+_RE_TEMPLATE_CHARS = re.compile(u'[{}]')
+_RE_TEMPLATE_CHARS_BYTES = re.compile(b'[{}]')
```

Comment:
I've been thinking about those character groups.
On the one hand, to be "real" Jinja, we need `{{`, `{#' or `{%' (and their counterparts), so the groups here also match on non-Jinja things. But then again, I wouldn't really care.
And we also do not really need to care for closing (the group above matches on either anyway).
So this could become
```suggestion
_RE_TEMPLATE_CHARS = re.compile(u'{')
_RE_TEMPLATE_CHARS_BYTES = re.compile(b'{')
```
at IMHO no (significant) downside, and with the upside that the matching is faster:
```console
% python3.12 /tmp/bench.py
0.19322030898183584
0.17506323801353574
0.09858699102187529
0.28751170000759885
```

Reply:
Yeah, the idea was to avoid matching things that obviously aren't templates (and can't concatenated to become templates), like IP addresses or hostnames, so that `ansible_host` and friends are only marked as unsafe if they contain potentially problematic things.

Whether the code should also check for `}` or not is :shrug:. I'll probably stick to the lazy option and not change it ;-) If it turns out to be too slow we can still reduce it to a simple `'{' in value` check.